cleanlab_studio/studio/studio.py

```diff
@@ -34,6 +35,8 @@ def upload_dataset(
         id_column: Optional[str] = None,
     ) -> str:
         ds = init_dataset_source(dataset, dataset_name)
+        if isinstance(ds, DataFrameDatasetSource) and dataset_name is None:
+            raise ValueError("Must provide dataset name if uploading from a DataFrame")
```

Comment:
won't we already have hit the assertion above?

Reply:
Oh yes good catch, I'll move this error message inside `init_dataset_source`